src/components/user/user.repository.ts

```diff
@@ -74,37 +75,54 @@ export class UserRepository extends DtoRepository(User) {
     ];
   }
 
-  private readonly roleProperties = (roles?: Role[]) =>
+  private readonly roleProperties = (roles?: Role[], baseNodeVar = 'user') =>
     (roles || []).flatMap((role) =>
-      property('roles', role, 'user', `role${role}`)
+      property('roles', role, baseNodeVar, `role${role}`)
     );
 
   async create(input: CreatePerson) {
     const id = await generateId();
-    const createdAt = DateTime.local();
+    const initialProps = {
+      realFirstName: input.realFirstName,
+      realLastName: input.realLastName,
+      displayFirstName: input.displayFirstName,
+      displayLastName: input.displayLastName,
+      phone: input.phone,
+      timezone: input.timezone,
+      about: input.about,
+      status: input.status,
+      title: input.title,
+      canDelete: true,
+    };
+    const baseNodeProps = { id };
+
     const query = this.db
       .query()
-      .create([
-        [
-          node('user', ['User', 'BaseNode'], {
-            id,
-            createdAt,
-          }),
-        ],
-        ...property('email', input.email, 'user', 'email', 'EmailAddress'),
-        ...property('realFirstName', input.realFirstName, 'user'),
-        ...property('realLastName', input.realLastName, 'user'),
-        ...property('displayFirstName', input.displayFirstName, 'user'),
-        ...property('displayLastName', input.displayLastName, 'user'),
-        ...property('phone', input.phone, 'user'),
-        ...property('timezone', input.timezone, 'user'),
-        ...property('about', input.about, 'user'),
-        ...property('status', input.status, 'user'),
-        ...this.roleProperties(input.roles),
-        ...property('title', input.title, 'user'),
-        ...property('canDelete', true, 'user'),
-      ])
-      .return('user.id as id')
+      .apply(
+        await createNode(User, {
+          initialProps,
+          baseNodeProps,
+        })
+      )
+      .apply((q) =>
```

Comment:
Cannot add `email` to `initialProps` because `getDbPropertyLabels` does not get `EmailAddress` label with empty db which means `devops@tsco.org` is not added to `EmailAddress` label when the server starts. I think we need to refactor `createNode` function?

Reply:
Just need to add
```ts
@DbLabel('EmailAddress')
readonly email: string
```
[here](https://github.com/SeedCompany/cord-api-v3/blob/develop/src/components/user/dto/user.dto.ts#L39)